internal/widget/base_renderer.go

```diff
@@ -0,0 +1,38 @@
+package widget
+
+import (
+	"image/color"
+
+	"fyne.io/fyne"
+	"fyne.io/fyne/theme"
+)
+
+// NewBaseRenderer creates a new BaseRenderer.
+func NewBaseRenderer(objects []fyne.CanvasObject) BaseRenderer {
```

Comment:
#827 says struct before constructor, but this looks okay too. I'd rather the codebase was consistent so should #827 be changed to this order?

Reply:
Oops … I wrote this code before #827. Because everybody™ agreed to #827 I already did some other refactorings with structs before constructors.
I will adjust this PR.